### PR TITLE
feat(navigation): allow passing QueryBuilder or QueryBuilderParams in `fetchNavigation` or `<ContentNavigation>`

### PR DESCRIPTION
**Pull request by harlan-zw**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The `ContentNavigation` component is typed to only allow you to provide the query as `QueryBuilderParams`. The DX of this isn't great as everywhere else in the doc a `QueryBuilder` is used with `queryContent`.

This is especially annoying when you want to show the `ContentNavigation` with a simple path prefix, as the user has to then figure out regex.

The `fetchContentNavigation` already supports a `QueryBuilder`, so it's seems like this was the initial expectation of this prop anyway.

I've added an example to highlight the use case

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/4.api/1.components/3.content-navigation.md [hunk 1]
```diff
@@ -43,3 +43,35 @@ The `empty`{lang=ts} slot can be used to display a default content before render
...
+By providing the `query` prop you can customise the content used for navigation.
+
+Here we pass along a `QueryBuilder` instance.
+
+```vue
+<script setup>
+const catsQuery = queryContent('cats')
+/*
+// If you'd prefer to pass along raw `QueryBuilderParams`:
+const contentQuery = {
```
Comment (Atinux) [inline docs/content/4.api/1.components/3.content-navigation.md:58]:
```suggestion
const catsQuery = {
```

File: docs/content/4.api/1.components/3.content-navigation.md [hunk 2]
```diff
@@ -43,3 +43,35 @@ The `empty`{lang=ts} slot can be used to display a default content before render
...
+    { _path: /^\/cats/ },
+  ],
+}
+*/
+</script>
+
+<template>
+<ContentNavigation v-slot="{ navigation }" :query="catsQuery">
+  <NuxtLink
+    v-for="link of navigation[0].children"
```
Comment (Atinux) [inline docs/content/4.api/1.components/3.content-navigation.md:69]:
```suggestion
    v-for="link of navigation"
```

For simplicity of example

**Review by Atinux — COMMENTED** [review]
Beautiful ❤️

**Comment by Tahul** [comment]
I love it as well! :)

Thanks a lot @harlan-zw 😄

**Review by Tahul — APPROVED** [review]

**Comment by Tahul** [comment]
Hey @harlan-zw :)

After a brief discussion with @farnabaz ; I remembered we had plans on supporting the same type of query format as `<ContentQuery />` has.

I will still merge this PR as we want to support both:

```vue
<script setup>
const queryBuilder = queryContent().where({ ... })
</script>

<template>
   <ContentNavigation :query="queryBuilder" />
   <!-- OR -->
   <ContentNavigation :where="{ ... }" />
</template>
```

**Comment by Tahul** [comment]
Just added a test to cover your changes @harlan-zw, thank you again for this!

**Comment by harlan-zw** [comment]
> Hey @harlan-zw :)
> 
> After a brief discussion with @farnabaz ; I remembered we had plans on supporting the same type of query format as `<ContentQuery />` has.
> 
> I will still merge this PR after applying some changes as we want to support both:
> 
> ```
> <script setup>
> const queryBuilder = queryContent().where({ ... })
> </script>
> 
> <template>
>    <ContentNavigation :query="queryBuilder" />
>    
>    <ContentNavigation :where="{ ... }" />
> </template>
> ```

This would definitely be the ideal case, having all components with the same prop API for querying would be a great DX improvement.

**Pull request merged**